internal/cron/monitor.go

```diff
@@ -80,10 +80,17 @@ func evaluateJob(ctx context.Context, logger *logrus.Logger, store *Store, job J
 				if err := store.FailJob(ctx, job.ID, -1, "machine destroyed before we could interpret the results"); err != nil {
 					log.WithError(err).Errorf("failed to update job %d status", job.ID)
 				}
-				return nil
+			} else {
+				log.WithError(err).Errorf("failed to get machine %s: %v", job.MachineID.String, err)
 			}
-			log.WithError(err).Errorf("failed to get machine %s: %v", job.MachineID.String, err)
```

Comment:
I'm not 100% sure if it deliberately proceeds in non-`404` cases, so let me know if it does. 🙂